subprojects/docs/src/docs/userguide/java.adoc

```diff
@@ -0,0 +1,879 @@
+
+You can learn more about these configurations and other aspects of building Java libraries in the plugin's chapter. Note that the Java Library Plugin automatically applies the standard Java Plugin as well.
+
+[[sec:building_java_applications]]
+=== Building Java applications
+
+Java applications packaged as a JAR aren't set up for easy launching from the command line or a desktop environment. The <<application_plugin,Application Plugin>> solves the command line aspect by creating a distribution that includes the production JAR, its dependencies and launch scripts Unix-like and Windows systems.
+
+See the plugin's chapter for more details, but here's a quick summary of what you get:
+
+ * `assemble` creates ZIP and TAR distributions of the application containing everything needed to run it
+ * A `run` task that starts the application from the build (for easy testing)
+ * Shell and Windows Batch scripts to start the application
+
+Note that you will need to explicitly apply the Java Plugin in your build script.
+
+[[sec:building_java_webapps]]
+=== Building Java web applications
+
+Java web applications can be packaged and deployed in a number of ways depending on the technology you use. For example, you might use https://projects.spring.io/spring-boot/[Spring Boot] with a fat JAR or a https://www.reactivemanifesto.org/[Reactive]-based system running on https://netty.io/[Netty]. Whatever technology you use, Gradle and its large community of plugins will satisfy your needs. Core Gradle, though, only directly supports traditional Servlet-based web applications deployed as WAR files.
+
+That support comes via the <<war_plugin,War Plugin>>, which automatically applies the Java Plugin and adds an extra packaging step that does the following:
+
+ * Copies static resources from _src/main/webapp_ into the root of the WAR
+ * Copies the compiled production classes into a _WEB-INF/classes_ subdirectory of the WAR
+ * Copies the library dependencies into a _WEB-INF/lib_ subdirectory of the WAR
+
+This is done by the `war` task, which effectively replaces the `jar` task — although that task remains — and is attached to the `assemble` lifecycle task. See the plugin's chapter for more details and configuration options.
+
+TODO:Check that we do in fact recommend the following. I noticed that the Web Quickstart does so.
+There is no core support for running your web application directly from the build, but we do recommend that you try the https://plugins.gradle.org/plugin/org.akhikhl.gretty[Gretty] community plugin, which provides an embedded Servlet container.
```

Comment:
@bmuschko I'm not sure what's going on with Gretty. It sounds as if it's being brought in house. Should I mention it here? Or leave that to the web-specific chapters?

Reply:
The original Gretty plugin has gotten little updates in the past year. Somebody forked the project and is now publishing to the plugin portal. I am confident that we will not make the plugin part of Gradle core. The forked plugin is this one: https://plugins.gradle.org/plugin/org.gretty.

We'll need to clean up the references to Gretty across the board in the user guide. I created an issue for it: https://github.com/gradle/gradle/issues/4902. For now you can keep the link you are using.

Reply:
I merged a PR that points to the new plugin: https://github.com/gradle/gradle/pull/4911. We should make the change here as well.